src/abstractdataframe/selection.jl

```diff
@@ -123,23 +116,12 @@ function normalize_selection(idx::AbstractIndex,
     return c => fun => newcol
 end
 
-function select_transform!(nc::Pair{Int, Pair{ColRename, Symbol}},
-                           df::AbstractDataFrame, newdf::DataFrame,
-                           transformed_cols::Dict{Symbol, Any}, copycols::Bool)
-    col_idx, (_, newname) = nc
-    # it is allowed to request column tranformation only once
-    @assert !hasproperty(newdf, newname)
-    newdf[!, newname] = copycols ? df[:, col_idx] : df[!, col_idx]
-    # mark that column transformation was applied
-    # nothing is not possible otherwise as a value in this dict
-    transformed_cols[newname] = nothing
-end
-
 function select_transform!(nc::Pair{<:Union{Int, AbstractVector{Int}},
                                     <:Pair{<:Union{Base.Callable, ByRow}, Symbol}},
                            df::AbstractDataFrame, newdf::DataFrame,
                            transformed_cols::Dict{Symbol, Any}, copycols::Bool)
     col_idx, (fun, newname) = nc
+    # it is allowed to request column tranformation only once
```

Comment:
I don't understand this comment. You mean that it's not allowed to specify multiple times the same transformation on the same column(s)?
```suggestion
    # a given transformation can only be requested once for each set of input(s)
```

Reply:
I mean that it is not allowed to make a transformation operation into the same "new column name" more than once.
This rule is handled elsewhere in the code and if such operation is passed to `select_transform!` then `@assert !hasproperty(newdf, newname)` should hold (so this line makes sure we do not have a bug in the code earlier).

By transformation we understand:
* an ordinary transformation
* column renaming
* single column selection (as we decided then we also do not want to allow for duplicates)

I will make a better comment.